gobblin-azkaban/src/main/java/gobblin/azkaban/AzkabanJobLauncher.java

```diff
@@ -125,12 +144,41 @@ public void run()
   public void cancel()
       throws Exception {
     try {
-      this.jobLauncher.cancelJob(this.jobListener);
+      cancelJob(this.jobListener);
     } finally {
-      this.closer.close();
+      try {
+        stop();
+      } finally {
+        close();
+      }
     }
   }
 
+  @Override
+  public void start() throws ApplicationException {
+    this.applicationLauncher.start();
```

Comment:
Yeah, but the `ServiceBasedAppLauncher.addService` method should be able to handle that. Although it may still be useful for other things so I will still add it.